tests/setup-validators.sh

```diff
@@ -0,0 +1,104 @@
+## testing the generated one with the default values provided by the module.
+
+# add in denom metadata for both native tokens
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.bank.denom_metadata += [{"name": "Work Token", "symbol": "WRK", "base": "worktoken", "description": "A test token for paying work", "denom_units": [{"denom": "worktoken", "exponent": 0}]},{"name": "Stake Token", "symbol": "STK", "base": "stake", "description": "A staking test token", "denom_units": [{"denom": "stake", "exponent": 0}]}]' /validator/config/genesis.json > /metadata-genesis.json
+docker cp /metadata-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/metadata-genesis.json
+
+# a 60 second voting period to allow us to pass governance proposals in the tests
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME jq '.app_state.gov.voting_params.voting_period = "60s"' /metadata-genesis.json > /edited-genesis.json
+docker cp /edited-genesis.json $STARTING_VALIDATOR_CONTAINER_NAME:/edited-genesis.json
+
+docker exec $STARTING_VALIDATOR_CONTAINER_NAME mv /edited-genesis.json /genesis.json
+
+# Copy genesis from starting node to host machine for gentx generation
+docker cp $STARTING_VALIDATOR_CONTAINER_NAME:/validator/config/genesis.json .
+
+# Sets up an arbitrary number of validators on a single machine by docker exec-ing on respective containers
+for i in $(seq 1 $NODES);
+do
+ARGS="$BASELEDGER_HOME --keyring-backend test"
+
+# Generate a validator key, orchestrator key for each validator
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys add $ARGS validator 2>> /validator-phrases
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys add $ARGS orchestrator 2>> /orchestrator-phrases
+
+VALIDATOR_KEY=$(docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys show validator -a $ARGS)
+ORCHESTRATOR_KEY=$(docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys show orchestrator -a $ARGS)
+
+# move the genesis in
+docker cp ./genesis.json $VALIDATOR_CONTAINER_BASE_NAME$i:/validator/config/genesis.json
+
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN add-genesis-account $ARGS $VALIDATOR_KEY $ALLOCATION
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN add-genesis-account $ARGS $ORCHESTRATOR_KEY $ALLOCATION
+
+# move the genesis back out
+docker cp $VALIDATOR_CONTAINER_BASE_NAME$i:/validator/config/genesis.json .
+
+done
+
+
+for i in $(seq 1 $NODES);
+do
+# move the genesis in
+docker cp ./genesis.json $VALIDATOR_CONTAINER_BASE_NAME$i:/validator/config/genesis.json
+
+ARGS="$BASELEDGER_HOME --keyring-backend test"
+ORCHESTRATOR_KEY=$(docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN keys show orchestrator -a $ARGS)
+
+VALIDATOR_CONTAINER_IP=$(docker inspect -f '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' $VALIDATOR_CONTAINER_BASE_NAME$i)
+
+docker exec $VALIDATOR_CONTAINER_BASE_NAME$i $BIN gentx $ARGS $BASELEDGER_HOME --moniker validator --chain-id=$CHAIN_ID --ip $VALIDATOR_CONTAINER_IP validator 500000000stake
```

Comment:
small thing but should they have different monikers?

Reply:
actually not sure, i thought moniker is local, we can change